engine/runtime/src/main/java/org/enso/interpreter/runtime/state/ExecutionEnvironment.java

```diff
@@ -53,49 +47,39 @@ public ExecutionEnvironment withContextDisabled(Atom context) {
   }
 
   private ExecutionEnvironment update(Atom context, boolean value) {
-    assert context.getConstructor().getType()
-        == EnsoContext.get(null).getBuiltins().context().getType();
-    int keyFound = -1;
-    for (int i = 0; i < keys.length; i++) {
-      if (keys[i].equals(context.getConstructor().getName())) {
-        keyFound = i;
-      }
-    }
-    String[] keys1;
-    Boolean[] permissions1;
-    if (keyFound != -1) {
-      keys1 = cloneArray(keys, new String[keys.length]);
-      permissions1 = cloneArray(permissions, new Boolean[keys.length]);
-      permissions1[keyFound] = value;
+    var contextBuiltin = EnsoContext.get(null).getBuiltins().context();
```

Comment:
It may be better to hide this into a node rather than ask for `get(null)`. A node like:
```java
class WithContextNode extends Node {
  public ExecutionEnvironment executeEnvironmentUpdate(
    ExecutionEnvironment current, Atom context, boolean enabled
  );
}
```

Reply:
Execution environment handling implemented via nodes in dd25a6f256918a675898e23ca95172ab3d5d463a